fabric-network/test/transaction.js

```diff
@@ -142,158 +166,105 @@ describe('Transaction', () => {
 		});
 	});
 
-	describe('#submit', () => {
-		let expectedProposal;
-
-		beforeEach(() => {
-			expectedProposal = {
-				fcn: transactionName,
-				txId: transaction.getTransactionID(),
-				chaincodeId: stubContract.getChaincodeId(),
-				args: []
-			};
-		});
-
-		it('rejects for non-string arguments', () => {
-			const promise = transaction.submit('arg1', 3.142, null);
-			return expect(promise).to.be.rejectedWith('"arg1", 3.142, null');
+	describe('#setEndorsingPeers', () => {
+		it('returns this', () => {
+			const result = transaction.setEndorsingPeers([]);
+			expect(result).to.equal(transaction);
 		});
+	});
 
-		it('sends proposal with no arguments', async () => {
-			await transaction.submit();
-			sinon.assert.calledWith(channel.sendTransactionProposal, sinon.match(expectedProposal));
+	describe('#setEndorsingOrganizations', () => {
+		it('returns this', () => {
+			const result = transaction.setEndorsingOrganizations([]);
+			expect(result).to.equal(transaction);
 		});
+	});
 
-		it('sends proposal with arguments', async () => {
-			const args = ['one', 'two', 'three'];
-			expectedProposal.args = args;
-			await transaction.submit(...args);
-			sinon.assert.calledWith(channel.sendTransactionProposal, sinon.match(expectedProposal));
+	describe('#getNetwork', () => {
+		it('returns network', () => {
+			transaction.network = network;
+			const result = transaction.getNetwork();
+			expect(result).to.equal(network);
 		});
+	});
 
-		it('returns null for no proposal response payload', async () => {
-			channel.sendTransactionProposal.resolves(noPayloadProposalResponses);
+	describe('#submit', () => {
```

Comment:
The original tests specify the behaviour of the `submit()` function for client code in various different scenarios. This behaviour should be consistent across releases and so all of these tests should remain. The new tests seems to drive certain implementation details and no longer describe behaviour.